Sources/KlaviyoCore/AppLifeCycleEvents.swift

```diff
@@ -9,18 +9,27 @@ import Combine
 import Foundation
 import UIKit
 
-enum LifeCycleErrors: Error {
+public enum LifeCycleErrors: Error {
     case invalidReachaibilityStatus
 }
 
-struct AppLifeCycleEvents {
-    var lifeCycleEvents: () -> AnyPublisher<KlaviyoAction, Never> = {
+public enum LifeCycleEvents {
+    case terminated
+    case forgrounded
+    case backgrounded
+    case reachabilityChanged(status: Reachability.NetworkStatus)
+}
+
+public struct AppLifeCycleEvents {
```

Comment:
In addition to moving this file out of `KlaviyoSwift` I also had to remove its dependence on `KlaviyoAction` that is part of the TCA architecture in `KlaviyoSwift`. To achieve this created a new type called `LifeCycleEvents` that is then mapped to `KlaviyoAction` in `KlaviyoSwift`